### PR TITLE
[MTDSA-1292] Amend & View Annual Gift Aid Relief

### PR DESCRIPTION
**Pull request by narendravijayarao**
Please check if this change includes (or even requires) the following:

 - [ ] unit tests
 - [ ] functional tests
 - [ ] audit events
 - [ ] RAML documentation
 - [ ] logging

### CODE REVIEW

File: app/uk/gov/hmrc/selfassessmentapi/resources/GiftAidPaymentsResource.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+object GiftAidPaymentsResource extends GiftAidPaymentsResource {
+  val giftAidPaymentsConnector = GiftAidPaymentsConnector
+}
+
+trait GiftAidPaymentsResource extends BaseResource {
+
+  val giftAidPaymentsConnector: GiftAidPaymentsConnector
+
+  def updatePayments(nino: Nino, taxYear: TaxYear): Action[JsValue] =
+    APIAction(nino, SourceType.GiftAidPayments).async(parse.json) { implicit request =>
```
Comment (MarkAKelly) [inline app/uk/gov/hmrc/selfassessmentapi/resources/GiftAidPaymentsResource.scala]: Can we add appropriate logging to these so we can track whats happening @narendravijayarao ?
Example: https://github.com/hmrc/vat-api/blob/master/app/uk/gov/hmrc/vatapi/resources/FinancialDataResource.scala#L50

File: func/uk/gov/hmrc/selfassessmentapi/resources/GiftAidPaymentsResourceFunctionalSpec.scala [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        .GiftAid
+        .updatePayments(nino, taxYear)
+        .when()
+        .put(Jsons.GiftAidPayments(100.00))
+        .at(s"/ni/$nino/gift-aid/$taxYear")
+        .thenAssertThat()
+        .statusIs(204)
+    }
+
+    "return code 409 when updating payments with invalid nino" in {
```
Comment (MarkAKelly) [inline func/uk/gov/hmrc/selfassessmentapi/resources/GiftAidPaymentsResourceFunctionalSpec.scala]: "return code 404..."

File: public/api/conf/1.0/application.raml [hunk 3]
```diff
@@ -1920,4 +1916,77 @@ traits:
...
+        description: !include docs/gift-aid-payments.md
+      /{taxYear}:
+        displayName: Gift Aid Payments for the tax year
+        uriParameters:
+          taxYear:
+            description: The tax year the data applies to
+            type: string
+            example: 2017-18
+        put:
+          displayName: Amend annual Gift Aid Tax Relief [test only]
```
Comment (MarkAKelly) [inline public/api/conf/1.0/application.raml]: 'Amend annual gift aid tax relief...'

File: public/api/conf/1.0/application.raml [hunk 4]
```diff
@@ -1920,4 +1916,77 @@ traits:
                       invalidTaxYear:
                         description: Invalid taxYear
                         value:
-                          code: TAX_YEAR_INVALID
\ No newline at end of file
+                          code: TAX_YEAR_INVALID
+    /gift-aid:
+      (annotations.group):
+        name: Gift Aid Payments
```
Comment (MarkAKelly) [inline public/api/conf/1.0/application.raml]: Gift aid

File: public/api/conf/1.0/application.raml [hunk 5]
```diff
@@ -1920,4 +1916,77 @@ traits:
...
+                      description: Amounts should be a non-negative number less than 99999999999999.98 with up to 2 decimal places
+                      value:
+                        code: INVALID_MONETARY_AMOUNT
+                    invalidTotalPayment:
+                      description: totalPayments cannot be less than the sum of totalOneOffPayments and payments of the nonUKCharityGift
+                      value:
+                        code: TOTAL_PAYMENTS_LESS
+            404:
+        get:
+          displayName: Get annual gift aid tax relief payments summary [test only]
```
Comment (MarkAKelly) [inline public/api/conf/1.0/application.raml]: 'View annual...'

Comment (narendravijayarao) [reply public/api/conf/1.0/application.raml]: Keeping as it is as discussed -- Get

**Pull request merged**